guide/getting-started-with-testcontainers-for-dotnet/index.adoc

```diff
@@ -0,0 +1,155 @@
+---
+title: "Getting started with Testcontainers for .NET"
+date: 2023-02-10T09:39:58+05:30
+draft: false
+description: This guide will help you to get started with Testcontainers for .NET by demonstrating how you can use PostgreSQL for testing.
+repo: https://github.com/testcontainers/tc-guide-getting-started-with-testcontainers-for-dotnet
+languages:
+  - C#
+tags:
+  - dotnet
+  - postgresql
+---
+:toc:
+:toclevels: 2
+:codebase: https://raw.githubusercontent.com/testcontainers/tc-guide-getting-started-with-testcontainers-for-dotnet/main
+
+Testcontainers is a testing library that provides easy and lightweight APIs for
+bootstrapping integration tests with real services wrapped in Docker
+containers. Using Testcontainers, you can write tests that talk to the same
+type of services you use in production, without mocks or in-memory services.
+
+[NOTE]
+If you are new to Testcontainers, please read
+https://testcontainers.com/guides/introducing-testcontainers[What is
+Testcontainers, and why should you use it?] to learn more about it.
+
+Let us look at how we can use Testcontainers to test a .NET application using a
+Postgres database.
+
+== Create a solution file with a source and test project
+
+Create a .NET source and test project from your favorite IDE. Once the projects
+are created, add the following dependencies to the source and test project.
+
+.Source project
+[source,xml,indent=0]
+----
+include::{codebase}/src/GettingStarted/GettingStarted.csproj[lines="6"]
+----
+
+.Test project
+[source,xml,indent=0]
+----
+include::{codebase}/tests/GettingStarted.Tests/GettingStarted.Tests.csproj[lines="6..8"]
+----
+
+We have added *Npgsql*, a Postgres ADO.NET Data Provider for talking to the
+Postgres database, and *xUnit.net* for testing our service.
+
+== Implement business logic
+
+We are going to create a `CustomerService` class to manage customer details.
+
+First, let us create a `Customer` type as follows:
+
+[source%nowrap,csharp]
+----
+include::{codebase}/src/GettingStarted/Customer.cs[]
+----
+
+Create a `DbConnectionProvider` class to hold ADO.NET connection parameters
+(connection string) and create a method to get a database connection, as follows:
+
+[source%nowrap,csharp]
+----
+include::{codebase}/src/GettingStarted/DbConnectionProvider.cs[]
```

Comment:
It should be "DBConnectionProvider.cs" (uppercase B).

Reply:
I favor `Db` to avoid [incontinent name](https://www.jetbrains.com/help/rider/InconsistentNaming.html) warnings in the IDE (UpperCamelCase / PascalCase). Our [modules](https://github.com/testcontainers/testcontainers-dotnet/tree/develop/src) follow the same style.

Reply:
The asciidoctor will try to fetch the content from that file and it is case sensitive, so it is failing to include the code snippet. To fix that you can rename the file to "DbConnectionProvider.cs".

Reply:
Good catch 🦅 Sorry, I misunderstood that you meant the actual file.